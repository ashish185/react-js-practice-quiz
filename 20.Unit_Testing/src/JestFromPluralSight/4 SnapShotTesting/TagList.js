import React from 'react';

export default ({ tags }) => {
    return (
        <div>
            {
                tags.map(tag => <code key={tag}>
                    {tag}
                </code>)
            }
        </div>
    );
};
