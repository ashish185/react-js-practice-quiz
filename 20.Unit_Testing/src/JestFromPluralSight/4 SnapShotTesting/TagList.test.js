import React from 'react'
import renderer from 'react-test-renderer';
import TagList from './TagList';

describe('The Tag component', () => {
    it("Renders as expected",()=>{
        const tree= renderer.create(<TagList tags={['css','bootstrap']}/>).toJSON();
        //Esko likhte hi ek new folder __snapshots__ k name se create ho jata h
        expect(tree).toMatchSnapshot();
    })
    
})
