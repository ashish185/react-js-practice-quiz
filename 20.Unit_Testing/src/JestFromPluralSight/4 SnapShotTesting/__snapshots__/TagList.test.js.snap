// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Tag component Renders as expected 1`] = `
<div>
  <code>
    css
  </code>
  <code>
    bootstrap
  </code>
</div>
`;
